src/mozillavpn.cpp

```diff
@@ -927,6 +919,14 @@ void MozillaVPN::mainWindowLoaded() {
 void MozillaVPN::onboardingCompleted() {
   SettingsHolder* settingsHolder = SettingsHolder::instance();
 
+  // Set glean based on data collection checkbox checked state
+  // Extracting this property from QML seems like the best approach to avoid
+  // creating a disposable member variable or setting
+  QObject* dataCollectionCheckbox =
```

Comment:
Nothing guarantees that this object is still there. I would use a setting flag, a signal, or something else here instead of qmlpath.